Core/Types/GameComparatorGRAS.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+
+namespace CKAN
+{
+    /// <summary>
+    /// Test to see if the user's game is "Generally Recognised As Safe" (GRAS) with a given mod,
+    /// with extra understanding of which KSP versions are "safe" (ie: 1.0.5 mostly works with 1.0.4 mods).
+    /// If the mod has `ksp_version_strict` set then this is identical to strict checking.
+    /// </summary>
+    public class GameComparatorGRAS : IGameComparator
+    {
+        static readonly GameComparatorStrict strict = new GameComparatorStrict();
+        static readonly KSPVersion v104 = new KSPVersion("1.0.4");
+
+        public bool Compatible(KSPVersion gameVersion, CkanModule module)
+        {
+            // If it's strictly compatible, then it's compatible.
+            if (strict.Compatible(gameVersion, module))
+                return true;
+
+            // If we're in strict mode, and it's not strictly compatible, then it's
+            // not compatible.
+            if (module.ksp_version_strict)
+                return false;
+
+            // Otherwise, check if it's "generally recognise as safe".
+
+            // If we're running KSP 1.0.5, then allow the mod to run if we would have
+            // considered it compatible under 1.0.4
+            if (gameVersion.Equals("1.0.5"))
```

Comment:
So hardcoding this is fine for now... but what I'd like to see in the long run is a probably something like a `config.json` in the top level of CKAN-meta that would contain dynamic configuration for clients that don't apply to any particular mod. So it could contain something like:

```json
{
  "ksp_version_compatability": {
    "1.0.4": [ "1.0.5" ]  
  }
}
```

Which would read as "Any mod that says its compatible with `1.0.4` is also compatible with `1.0.5` by the default (GRAS) scheme.

By making it part of the metadata we can update it through PRs and make sure older clients get the necessary information without relying upon people upgrading the client.